types/dns_service_address.pp

```diff
@@ -0,0 +1,7 @@
+# a type to describe the dns_service_address
+type K8s::Dns_service_address = Variant[
```

Comment:
Should really be something like `K8s::Ip_addresses` since it's just a list of one or more subnet-free IP Addresses, there's nothing requiring it to be used for the DNS service